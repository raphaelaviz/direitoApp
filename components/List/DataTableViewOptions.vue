<script setup lang="ts">
import type { Table } from '@tanstack/vue-table'
import { computed } from 'vue'
import type { LawsuitType } from '@/lib/utils'

import { Button } from '@/components/ui/button'
import { Eye } from 'lucide-vue-next';


interface DataTableViewOptionsProps {
  table: Table<LawsuitType>
}

const props = defineProps<DataTableViewOptionsProps>()

const columns = computed(() => props.table.getAllColumns()
  .filter(
    column =>
      typeof column.accessorFn !== 'undefined' && column.getCanHide(),
  ))
</script>

<template>
  <DropdownMenu>
    <CustomTooltip content="View">
    <DropdownMenuTrigger as-child>
      <Button
        variant="outline"
        size="sm"
        class="ml-auto hidden h-8 lg:flex"
      >
      <span class="sr-only">View</span>
       <Eye/>
      </Button>
    </DropdownMenuTrigger>
    <DropdownMenuContent align="center" class="w-[150px]">
      <DropdownMenuLabel>View</DropdownMenuLabel>
      <DropdownMenuSeparator />

      <DropdownMenuCheckboxItem
        v-for="column in columns"
        :key="column.id"
        class="capitalize"
        :checked="column.getIsVisible()"
        @update:checked="(value) => column.toggleVisibility(!!value)"
      >
      <!-- TODO: delete this conditional rendering after changing the tables names  -->
      {{ column.id === 'createdAt' ? 'Created at' : column.id }}

      </DropdownMenuCheckboxItem>
    </DropdownMenuContent>
  </CustomTooltip>
  </DropdownMenu>
</template>
