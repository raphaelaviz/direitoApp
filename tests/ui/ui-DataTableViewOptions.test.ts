import DataTableViewOptions from "../../components/List/DataTableViewOptions.vue"
import { it, expect, describe, test } from 'vitest'
import { renderSuspended } from '@nuxt/test-utils/runtime'

// last snap shot: 17/04

it('matches snapshot', async () => {
  const wrapper = await renderSuspended(DataTableViewOptions)
  expect(wrapper).toMatchSnapshot()
})