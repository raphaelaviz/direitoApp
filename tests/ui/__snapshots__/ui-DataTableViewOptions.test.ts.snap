// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
{
  "baseElement": <body>
    <div
      data-v-app=""
      id="__nuxt"
    >
      <!---->
    </div>
    <div>
      <div
        id="test-wrapper"
      >
        
        
        
        
        
        
        
        
        
        
        <button
          aria-expanded="false"
          aria-haspopup="menu"
          class="items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground rounded-md px-3 outline-none ml-auto hidden h-8 lg:flex"
          data-state="closed"
          id="radix-vue-dropdown-menu-trigger-2"
          type="button"
        >
          
          <span
            class="sr-only"
          >
            View
          </span>
          <svg
            class="lucide lucide-eye-icon"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 12s3-7 10-7 10 7 10 7-3 7-10 7-10-7-10-7Z"
            />
            <circle
              cx="12"
              cy="12"
              r="3"
            />
          </svg>
          
        </button>
        <!--teleport start-->
        <!--teleport end-->
        
        <!--teleport start-->
        <!--teleport end-->
        
        
        
        
        
        
        
        
        
      </div>
    </div>
    
    
    
    <!---->
    
    
    
    
    
    
    <!---->
    
    
    
  </body>,
  "container": <div>
    <div
      id="test-wrapper"
    >
      
      
      
      
      
      
      
      
      
      
      <button
        aria-expanded="false"
        aria-haspopup="menu"
        class="items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground rounded-md px-3 outline-none ml-auto hidden h-8 lg:flex"
        data-state="closed"
        id="radix-vue-dropdown-menu-trigger-2"
        type="button"
      >
        
        <span
          class="sr-only"
        >
          View
        </span>
        <svg
          class="lucide lucide-eye-icon"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 12s3-7 10-7 10 7 10 7-3 7-10 7-10-7-10-7Z"
          />
          <circle
            cx="12"
            cy="12"
            r="3"
          />
        </svg>
        
      </button>
      <!--teleport start-->
      <!--teleport end-->
      
      <!--teleport start-->
      <!--teleport end-->
      
      
      
      
      
      
      
      
      
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
